pkg/bloomcompactor/chunkcompactor.go

```diff
@@ -22,7 +22,7 @@ import (
 )
 
 type compactorTokenizer interface {
-	PopulateSeriesWithBloom(bloom *v1.SeriesWithBloom, chunks []chunk.Chunk) error
+	PopulateSeriesWithBloom(bloom *v1.SeriesWithBloom, chunkBatchesIterator v1.Iterator[[]chunk.Chunk]) error
```

Comment:
yes, it will do the same... but I do not believe that we win anything by changing it this way... 
it just increases the complexity of the iterator. 
let's say if we failed to download the batch, we will report an error, but this error is not connected to this particular chunk is returned by `At()` function. so, it might confuse somebody in the future... 
not really against it, but it looks almost the same.

Reply:
wdyt?

Reply:
Sure, not a strong opinion!